ruby/neo4j/driver/internal/internal_session.rb

```diff
@@ -48,11 +48,11 @@ def write_transaction(**config, &block)
       end
 
       def execute_read(**config, &block)
-        transaction(AccessMode::READ, **config, &block)
+        transaction(AccessMode::READ, **config) { |tx| block.call(DelegatingTransaction.new(tx)) }
       end
 
       def execute_write(**config, &block)
-        transaction(AccessMode::WRITE, **config, &block)
+        transaction(AccessMode::WRITE, **config) { |tx| block.call(DelegatingTransaction.new(tx)) }
```

Comment:
Now that it has been simplified, it becomes more obvious that the duplication needs to be fixed.
Introduce a parameter in `transaction` method or maybe better a private `delegating_transaction` method which delegates to `transaction` with modified block.